### PR TITLE
Add code to generate the MPAS initial conditions from the RAP rotated latlon grid fields.

### PR DESCRIPTION
**Pull request by hu5970**
The new code for processing the RAP rotated latlon grid was added from Larissa's repository:

https://github.com/LarissaReames-NOAA/MPAS-Model.git
Branch NSSL_develop_RAPinit

The parallel tests show the new code using RAP rotated latlon grid file can generate mostly the same results as using RAP 130 grid files.

### CODE REVIEW

**Comment by hu5970** [comment]
@jderrico-noaa Could you add Larissa Reames as a reviewer ([larissa.reames@noaa.gov](https://github.com/LarissaReames-NOAA) )? Thanks, Ming

**Comment by jderrico-noaa** [comment]
Are there changes here besides what's in Larissa's branch?  Or is what's in this PR identical to the NSSL_develop_RAPinit branch?

**Comment by hu5970** [comment]
All change from Larissa's branch. But on top of GSL repository.

**Comment by jderrico-noaa** [comment]
In that case we should do this PR as a merge from Larissa's branch directly so we can track the full history.  If it's easier I can create the PR and resolve any conflicts according to what's in this one, and then credit you and Larissa and add you both as reviewers?

**Comment by hu5970** [comment]
@jderrico-noaa The code was from Larissa's branch. But the code has been merged and tested on top of the current GSL repository. Larissa's branch is based on different MPAS version. It is easier and safer to merge from my PR.

**Comment by hu5970** [comment]
@jderrico-noaa Please add reviewers and process this PR. Let me know if you have any question on this PR.

**Comment by jderrico-noaa** [comment]
@hu5970  The way the NOAA-GSL GitHub organization permissions are currently configured I can't add anyone outside of NOAA-GSL.  I opened a ticket with ITS when this PR was opened and they're still working on it--for now it seems like the fastest solution is to add Larissa to NOAA-GSL, but moving forward we will need a better solution for external collaborators

**Comment by hu5970** [comment]
@jderrico-noaa Thanks for updates. NO surprise there are extra efforts to work with external collaborators.

**Review by joeolson42 — APPROVED** [review]

File: src/core_init_atmosphere/mpas_init_atm_cases.F [hunk 1]
```diff
@@ -2854,7 +2854,7 @@ subroutine init_atm_case_gfs(block, mesh, nCells, nEdges, nVertLevels, fg, state
       character (len=StrKIND), pointer :: config_interface_projection
 
       real (kind=RKIND) :: truelat1, truelat2, xlonc, cone
-
+      real (kind=RKIND) :: temp
```
Comment (AndersJensen-NOAA) [inline src/core_init_atmosphere/mpas_init_atm_cases.F:2857]: `temp` is not the best variable name

**Review by AndersJensen-NOAA — APPROVED** [review]

File: src/core_init_atmosphere/mpas_init_atm_cases.F (hunk 1 above)
Comment (hu5970) [reply src/core_init_atmosphere/mpas_init_atm_cases.F:2857]: Agree. But I cannot figure out a better name.

**Comment by hu5970** [comment]
@jderrico-noaa Joe and Anders approved this PR. Please check it. Thanks, Ming

**Pull request merged**